### DISCUSSION TITLE
Needed:  crowdstrike agent uninstall method for hard cases

### DISCUSSION THREAD

**Discussion by RacerXNumber9** [CrowdStrike/psfalcon#338] [General]
I found that psfalcon has a command to get uninstall token:  Get-FalconUninstallToken
...but it requires -Id  parameter  - **how/where can I get the accurate Id   data for parameter ?**   
Also - our company uses crowdstrike globally and sure enough, we have 'hard cases'  where the agent
gets 'messed up'  on client machines -  won't upgrade properly etc. and we need to do a FORCE UNINSTALL 
 ...but is never easy
If we had a **force uninstall method** for crowdstrike agent on client machines  windows and linux ...that would be 
super helpful -  our Security team currently has to go through a LOT of pain and suffering to simply get agent
off/uninstalled on clients so that latest agent can be installed.

**Comment by bk-cs** [comment]
There are two ways to get the required identifier for `Get-FalconUninstallToken`:

1. If the host is still properly reporting into the Falcon console, use `Get-FalconHost` and a filtered search to find it
2. If the host is *not* reporting into the Falcon console you can check the local host for its identifier

CrowdStrike support can assist in helping you locate the identifier on a local machine, either through PowerShell or the Windows command prompt, or the required tool if using macOS/Linux.

Assuming you get the uninstall token after that, you can use a script or the CsUninstallTool for Windows to remove the sensor.
\[ [EU-1](https://falcon.eu-1.crowdstrike.com/support/tool-downloads) | [US-1](https://falcon.crowdstrike.com/support/tool-downloads) | [US-2](https://falcon.us-2.crowdstrike.com/support/tool-downloads) | [US-GOV-1](https://falcon.laggar.gcw.crowdstrike.com/support/tool-downloads) \]
